src/bindings/python/flux/job/info.py

```diff
@@ -541,6 +541,59 @@ def zero_remove(key):
 
         return result
 
+    @memoized_property
+    def inactive_reason(self):
+        """
+        Generate contextual exit reason based on how the job ended
+        """
+        state = str(self.state)
+        if state != "INACTIVE":
+            return ""
+        result = str(self.result)
+        if result == "CANCELED":
+            if (
+                self.exception.occurred
+                and self.exception.type == "cancel"
+                and self.exception.note
+            ):
+                return f"Canceled: {self.exception.note}"
+            else:
+                return "Canceled"
+        elif result == "FAILED":
+            # exception.type == "exec" is special case, handled by returncode
+            if (
+                self.exception.occurred
+                and self.exception.type != "exec"
+                and self.exception.severity == 0
+            ):
+                note = None
+                if self.exception.note:
+                    note = f" note={self.exception.note}"
+                return f'Exception: type={self.exception.type}{note or ""}'
+            elif self.returncode > 128:
+                signum = self.returncode - 128
+                # strsignal() is only available in Python 3.8 and up.
+                # flux-core's minimum is 3.6.  Use compat library
+                from flux.compat36 import strsignal
```

Comment:
I got a vermin failure when I tried this previously, lemme give it another shot.  Perhaps the import in the middle of the file causes part of the problem.

Reply:
going into vermin's code, it seems like this is just a hard require

```
    "signal.strsignal": (None, (3, 8)),
```

BUT I cannot generate this error message on a local install of vermin with 1.5.1, wondering if this could be worked around with a newer version of vermin.  playing around.

Reply:
well that's a bummer. Vermin makes it so you can't support multiple versions of Python? Do we really need the vermin checks? Our CI already runs all tests with multiple versions of Python.

Though maybe doing the ImportError thing is considered bad practice by Python folks? I'm having trouble forming an opinion now. Sorry for putting you through that.